### DISCUSSION TITLE
Bug in the code

### DISCUSSION THREAD

**Discussion by theo2021** [astra-vision/rain-rendering#9] [Ideas] [answered]
This for loop: https://github.com/cv-rits/rain-rendering/blob/6365f2077764320f0a1bd5244eda2a5489225230/main_threaded.py#L114, should be changed to
`for frame_start in range(args.frame_start, args.frame_end, frames_per_thread):` so that it starts from the frame start and not 0. Moreover, I don't really understand the use of this: https://github.com/cv-rits/rain-rendering/blob/6365f2077764320f0a1bd5244eda2a5489225230/main_threaded.py#L118. Cheers

**Comment by rdecharette** [comment, marked as answer]
Hello Theo. 
True `args.frame_start` was missing, it's quite unlikely to cause issue but I fixed it. 

If I recall correctly,
https://github.com/cv-rits/rain-rendering/blob/9e994b0cd7e5e210189821345f6655b21c79ca48/main_threaded.py#L118
is for the case where multiple scenes should be rendered to split them in different threads. Bare in mind this file is snippet - not optimized - to automatize rendering on multiple sequences.

Best

**Reply by theo2021** [reply]
Yes, but if one scene is selected, multiple threads are still being created, rendering the same images many times. In my code, I removed it and performed a queue of intensities

**Reply by rdecharette** [reply]
That's quite not expected indeed. Thanks for the bug heads up.
